src/abstraction/transient.rs

```diff
@@ -207,6 +278,86 @@ impl TransientKeyContext {
         Ok(key)
     }
 
+    /// Encrypt a message with an existing key.
+    ///
+    /// Takes the key as a parameter, encrypts the message and returns the ciphertext
+    ///
+    /// # Errors
+    /// * errors are returned if any method calls return an error: `Context::context_load`,
+    /// `Context::rsa_encrypt`, `Context::flush_context`, `TransientKeyContext::set_session_attrs`
+    /// `Context::set_handle_auth`
+    pub fn rsa_encrypt(
+        &mut self,
+        key_context: TpmsContext,
+        key_auth: Option<Auth>,
+        message: PublicKeyRSA,
+    ) -> Result<PublicKeyRSA> {
+        self.set_session_attrs()?;
+        let key_handle = self.context.context_load(key_context)?;
+        if let Some(key_auth_value) = key_auth {
+            self.context
+                .tr_set_auth(key_handle, &key_auth_value)
+                .or_else(|e| {
+                    self.context.flush_context(key_handle)?;
+                    Err(e)
+                })?;
+        }
+        let scheme = AsymSchemeUnion::RSAOAEP(HashingAlgorithm::Sha256).get_rsa_decrypt_struct();
```

Comment:
Hmm, should the hash algorithm be chosen by the user? I don't know how widespread use of non-SHA256 is.

Reply:
If we keep it like this, it's worth adding a line in the docs above to mention the encryption scheme (since it's not chosen when the key is created, like with signing).